lib/SpeculationModules/PDGBuilder.cpp

```diff
@@ -35,6 +35,9 @@ using namespace llvm;
 using namespace llvm::noelle;
 using namespace liberty;
 
+unsigned int edge_count = 0;

```

Comment:
OK, the logic of this file needs to be removed to Planner in CPF as an optional feature. 
You need to do the following:
- Create a `cl::opt` that says something like ValidityCheck etc
- When the option is on, check the analysis results against SLAMP, and/or check SLAMP against LAMP
- When doing checks, in function `parallelizeLoop` of Planner.cpp, after you get the pdg, use SLAMP and LAMP oracle AA as Remediators through `memdep` API
- Dump the differences out